types/structure_description.h

```diff
@@ -171,8 +180,8 @@ struct StructureDescription
         if (fields.count(name.c_str()))
             throw ML::Exception("field '" + name + "' added twice");
 
-        fieldNames.push_back(name);
-        const char * fieldName = fieldNames.back().c_str();
+        fieldNames.emplace_back(::strdup(name.c_str()));
```

Comment:
Actually strdup is not part of std, due to it being from POSIX rather than the ISO C std. It is not even forwarded by the gnu c++ lib. Therefore using "std::" would cause compilation failures (I checked).